packages/app/e2e/helpers.js

```diff
@@ -0,0 +1,12 @@
+exports.getE2eTestProject = function getE2eTestProject() {
+  return 'react-native-firebase-testing';
+};
+
+// Android requires a re-mapped hostname during testing sometimes
+exports.getE2eEmulatorHost = function getE2eEmulatorHost() {
+  // This appears to work from localhost even on android. Different execution context?
+  // if (platform === 'android') {
+  //   return '10.0.2.2';
+  // }
+  return 'localhost';
```

Comment:
This works as localhost due to; https://github.com/invertase/react-native-firebase/pull/4552/files#diff-17651da2be070111f5b3f961a4e6fda957ed253e301e14f8a1da4a5ce95c44efR343-R351

We dod this on all emulator apis for RNFB so the users don't need to worry about it

Reply:
Yeah, I got that part from the functions emulator implementation and it works great. This is separate though. This helper function is returning a string without any intervening translation to a higher layer of helper function which constructs a raw URL out of it with no other translation, and only localhost works. It makes me think the e2e javascript is running in the context of jet on the host as opposed to the react-native JS executor in the emulator. I verified that you need 10.0.2.2 if you try it on the emulator (using `adb shell` and then `nc` in that shell to pretend I was a web client).

TL:DR; which you probably know: the e2e javascript is calling functions in the emulator but the regular javascript, is it executing in jet on the host?

Reply:
Oh ye sorry I misread this, e2e JS does run in Node.js on the host machine and not the emulator - so this makes sense.

Reply:
perfect, I will update that comment as "where things run" is vital to know, especially since we (as mentioned) do the translation to 10.0.2.2 in other places.

Side note: I do the equivalent of rebase -i for these PRs so the commits can stay separate, I've taken care that they'll even survive (with correct success/fail) if applied partially but in sequence so git bisect will work (something we learned to be careful about in AnkiDroid). Not the standard I would hold most contributors too but I think these commits serve best if merged in separated, similar to my iosv7/androidv26 forward port